### PR TITLE
Update BSPs for ecc memory scrub. Bump submodules.

### PR DESCRIPTION
**Pull request by NandkumarJoshi**

### CODE REVIEW

**Comment by nategraff-sifive** [comment]
So the error that happened here was because the ldscript generator didn't guarantee reproducibility. I've fixed the problems and already bumped the generators past these commits, so you can probably get away with reducing this PR to just bump freedom-metal.

**Comment by bsousi5** [comment]
You also need to bump freedom-e-sdk in wit-manifest.json. I merged a fix to devicetree-overlay-generator this afternoon.

**Comment by NandkumarJoshi** [comment]
> So the error that happened here was because the ldscript generator didn't guarantee reproducibility. I've fixed the problems and already bumped the generators past these commits, so you can probably get away with reducing this PR to just bump freedom-metal.

Thanks for the fix ! Updated this PR to pickup latest freedom-metal, device-tools, ld-generator. Regenerate BSPs.

File: debug.mk [hunk 1]
```diff
@@ -9,6 +9,5 @@ RISCV_CXXFLAGS += -O0
 
 # Enable debug
 RISCV_ASFLAGS += -g
-RISCV_CFLAGS += -g
-RISCV_CXXFLAGS += -g
-
+RISCV_CFLAGS += -g -mno-relax
+RISCV_CXXFLAGS += -g -mno-relax
```
Comment (bsousi5) [inline debug.mk:13]: Since this change is limited to only with debug build, I will allow this for now. We would not want to do this across the board.

**Pull request merged**